### PR TITLE
add integrations datasource back in

### PR DESCRIPTION
**Pull request by davidbloss**
## Issues

https://github.com/OpsLevel/team-platform/issues/289

## Changelog

Changes:
- Add `integrations` datasource

- [X] List your changes here
- [X] Make a `changie` entry

## Tophatting

Using this file:
```terraform
# main.tf
data "opslevel_integration" "by_id" {
  filter {
    field = "id"
    value = "Z2lkOi8vb3BzbGV2ZWwvSW50ZWdyYXRpb25zOjpFdmVudHM6OkRvY3VtZW50czo6QXBpRG9jSW50ZWdyYXRpb24vMTc4Nw"
  }
}
data "opslevel_integration" "by_name" {
  filter {
    field = "name"
    value = "Terraform"
  }
}

data "opslevel_integrations" "all" {
}

output "all" {
  value = data.opslevel_integrations.all
}

output "by_id" {
  value = data.opslevel_integration.by_id
}

output "by_name" {
  value = data.opslevel_integration.by_name
}
```

Run `terraform plan`
```terraform
data.opslevel_integrations.all: Reading...
data.opslevel_integration.by_name: Reading...
data.opslevel_integration.by_id: Reading...
data.opslevel_integration.by_name: Read complete after 1s [id=Z2lkOi8vb3BzbGV2ZWwvSW50ZWdyYXRpb25zOjpUZXJyYWZvcm1JbnRlZ3JhdGlvbi8xMjI2]
data.opslevel_integration.by_id: Read complete after 1s [id=Z2lkOi8vb3BzbGV2ZWwvSW50ZWdyYXRpb25zOjpFdmVudHM6OkRvY3VtZW50czo6QXBpRG9jSW50ZWdyYXRpb24vMTc4Nw]
data.opslevel_integrations.all: Read complete after 1s

Changes to Outputs:
  + all     = {
      + integrations = [
          + {
              + id   = "Z2lkOi8vb3BzbGV2ZWwvSW50ZWdyYXRpb25zOjpHaXRodWJJbnRlZ3JhdGlvbi8xNjk"
              + name = "GitHub - suraj-testing2"
            },
          + {
              + id   = "Z2lkOi8vb3BzbGV2ZWwvSW50ZWdyYXRpb25zOjpFdmVudHM6OkRlcGxveUludGVncmF0aW9uLzE4MA"
              + name = "Chris' Deploy"
            },
          + {
              + id   = "Z2lkOi8vb3BzbGV2ZWwvSW50ZWdyYXRpb25zOjpHaXRodWJJbnRlZ3JhdGlvbi8xODM"
              + name = "GitHub - OL-Templates"
            },
          ... # edited here to not include allll the integrations
          + {
              + id   = "Z2lkOi8vb3BzbGV2ZWwvSW50ZWdyYXRpb25zOjpHaXRodWJJbnRlZ3JhdGlvbi80OTAy"
              + name = "GitHub - eapache-opslevel"
            },
        ]
    }
  + by_id   = {
      + filter = {
          + field = "id"
          + value = "Z2lkOi8vb3BzbGV2ZWwvSW50ZWdyYXRpb25zOjpFdmVudHM6OkRvY3VtZW50czo6QXBpRG9jSW50ZWdyYXRpb24vMTc4Nw"
        }
      + id     = "Z2lkOi8vb3BzbGV2ZWwvSW50ZWdyYXRpb25zOjpFdmVudHM6OkRvY3VtZW50czo6QXBpRG9jSW50ZWdyYXRpb24vMTc4Nw"
      + name   = "API Docs"
    }
  + by_name = {
      + filter = {
          + field = "name"
          + value = "Terraform"
        }
      + id     = "Z2lkOi8vb3BzbGV2ZWwvSW50ZWdyYXRpb25zOjpUZXJyYWZvcm1JbnRlZ3JhdGlvbi8xMjI2"
      + name   = "Terraform"
    }

You can apply this plan to save these new output values to the Terraform state, without changing any real infrastructure.
```

### CODE REVIEW

File: opslevel/datasource_opslevel_integrations.go
Comment (taimoorgit) [inline opslevel/datasource_opslevel_integrations.go:1]: This file should be deleted and the plural datasource file should contain `_all`

File: opslevel/datasource_opslevel_integration.go [hunk 1]
```diff
@@ -23,18 +23,29 @@ type IntegrationDataSource struct {
...
+		Computed:    true,
+	},
+	"name": schema.StringAttribute{
+		Description: "The name of the Integration.",
+		Computed:    true,
+	},
+}
+
+// IntegrationDataSourceWithFilterModel describes the data source data model.
+type IntegrationDataSourceWithFilterModel struct {
```
Comment (taimoorgit) [inline opslevel/datasource_opslevel_integration.go:38]: Should be unexported along with `NewIntegrationDataSourceModel`

File: opslevel/datasource_opslevel_integrations.go [hunk 2]
```diff
@@ -1,52 +1,91 @@
...
+	var planModel, stateModel IntegrationDataSourcesAllModel
+
+	// Read Terraform configuration data into the model
+	resp.Diagnostics.Append(req.Config.Get(ctx, &planModel)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	integrations, err := i.client.ListIntegrations(nil)
+	if err != nil || integrations == nil || integrations.Nodes == nil {
```
Comment (taimoorgit) [inline opslevel/datasource_opslevel_integrations.go:80]: Not fault of this PR but it's painful that we have to perform these checks...

**Review by taimoorgit — APPROVED** [review]

File: opslevel/datasource_opslevel_integrations.go (hunk 2 above)
Comment (davidbloss) [reply opslevel/datasource_opslevel_integrations.go]: 💯 agreed

**Pull request merged**